src/OSPSuite.Infrastructure.Import/Services/Importer.cs

```diff
@@ -146,6 +155,54 @@ public MappingProblem CheckWhetherAllDataColumnsAreMapped(IReadOnlyList<ColumnIn
                .ToList()
          };
       }
+
+      public IReadOnlyList<DataRepository> DataSourceToDataSets(IDataSource dataSource, IReadOnlyList<MetaDataCategory> metaDataCategories,
+         DataImporterSettings dataImporterSettings, string id)
+      {
+         var dataRepositories = new List<DataRepository>();
+
+         for (var i = 0; i < dataSource.DataSets.SelectMany(ds => ds.Data).Count(); i++)
+         {
+            var dataRepo = _dataRepositoryMapper.ConvertImportDataSet(dataSource.DataSetAt(i));
+            dataRepo.ConfigurationId = id;
+            var moleculeDescription =
+               (metaDataCategories?.FirstOrDefault(md => md.Name == dataImporterSettings.NameOfMetaDataHoldingMoleculeInformation)?.ListOfValues
+                  .FirstOrDefault(v => v.Key == dataRepo.ExtendedPropertyValueFor(dataImporterSettings.NameOfMetaDataHoldingMoleculeInformation)))
+               ?.Value;
+            var molecularWeightDescription = dataRepo.ExtendedPropertyValueFor(dataImporterSettings.NameOfMetaDataHoldingMolecularWeightInformation);
+
+            if (moleculeDescription != null)
+            {
+               if (string.IsNullOrEmpty(molecularWeightDescription))
+               {
+                  molecularWeightDescription = moleculeDescription;
+               }
+               else
+               {
+                  double.TryParse(moleculeDescription, out var moleculeMolWeight);
+                  double.TryParse(molecularWeightDescription, out var molWeight);
+
+                  if (!ValueComparer.AreValuesEqual(moleculeMolWeight, molWeight))
+                  {
+                     throw new InconsistentMoleculeAndMoleWeightException();
+                  }
+               }
+            }
+
+            if (!string.IsNullOrEmpty(molecularWeightDescription))
+            {
+               if (double.TryParse(molecularWeightDescription, out var molWeight))
+               {
+                  dataRepo.AllButBaseGrid().Each(x => x.DataInfo.MolWeight = molWeight);
+               }
+            }
+
+            dataRepo.ExtendedProperties.Remove(dataImporterSettings.NameOfMetaDataHoldingMolecularWeightInformation);
```

Comment:
This is removed because after we create the dataRepository the MolWeight exists in its properties. I f we keep the extended property, then the MolWeight is duplicated.

Reply:
Please add a comment to explain this. This is typically the very few places where comment in the code are helpful